applications/salk-portal/src/models/OverlayMetadata.ts

```diff
@@ -1,9 +1,12 @@
+
 export default class OverlayMetadata {
```

Comment:
@afonsobspinto can't we come up with a better way of doing this?
I mean react has a useEffect with dependencies why do we need to create our "own" dependency tree....

Reply:
I made the overlay handling generic by using a sort of declarative strategy. I like it because I think it facilitates the addition of new overlays. 

Opposed to having to create specific handlers for each new overlay we might want, with the above strategy if we decide to add a new overlay I just need to 'declare' the new one in the Overlays constants and everything will work. 

Given the above explanation, please let me know if you still want me to change this approach, to which approach would you like me to change (the specific handlers for each overlay?) and if possible why would you prefer that solution. cc @zsinnema

Reply:
@afonsobspinto I put this PR back to draft since you have to change the density map generation according to our chat last Friday with Paolo: generate the image per population on population creation/updating and then merge them client side